test/Npgsql.Tests/NpgsqlParameterCollectionTests.cs

```diff
@@ -0,0 +1,256 @@
+using System;
+using NpgsqlTypes;
+using NUnit.Framework;
+
+namespace Npgsql.Tests
+{
+    // This test class has global effects on case sensitive matching in param collection.
+    [NonParallelizable]
+    [TestFixture(CompatMode.CaseSensitive)]
+    [TestFixture(CompatMode.CaseInsensitive)]
+    public class NpgsqlParameterCollectionTests
+    {
+        readonly CompatMode _compatMode;
+        const int LookupThreshold = NpgsqlParameterCollection.LookupThreshold;
+
+        /// <summary>
+        /// Test which validates that Clear() indeed cleans up the parameters in a command so they can be added to other commands safely.
+        /// </summary>
+        [Test]
+        public void NpgsqlParameterCollectionClearTest()
+        {
+            var p = new NpgsqlParameter();
+            var c1 = new NpgsqlCommand();
+            var c2 = new NpgsqlCommand();
+            c1.Parameters.Add(p);
+            Assert.AreEqual(1, c1.Parameters.Count);
+            Assert.AreEqual(0, c2.Parameters.Count);
+            c1.Parameters.Clear();
+            Assert.AreEqual(0, c1.Parameters.Count);
+            c2.Parameters.Add(p);
+            Assert.AreEqual(0, c1.Parameters.Count);
+            Assert.AreEqual(1, c2.Parameters.Count);
+        }
+
+                [Test]
```

Comment:
nit: move the attribute a little bit to the left

Reply:
Just checking if you're paying attention :wink: